### PR TITLE
principal_nodal_stress

### PR DESCRIPTION
**Pull request by 1990chs**
try to improve principal_nodal_stress() API to add nodes option.

### CODE REVIEW

File: ansys/mapdl/reader/rst.py [hunk 1]
```diff
@@ -2413,7 +2413,7 @@ def element_solution_data(self, rnum, datatype, sort=True, **kwargs):
 
         return enum, element_data, enode
 
-    def principal_nodal_stress(self, rnum):
+    def principal_nodal_stress(self, rnum, nodes = None):
```
Comment (akaszynski) [inline ansys/mapdl/reader/rst.py:2416]:
```suggestion
    def principal_nodal_stress(self, rnum, nodes=None):
```

File: ansys/mapdl/reader/rst.py [hunk 2]
```diff
@@ -2458,7 +2458,7 @@ def principal_nodal_stress(self, rnum):
 
         """
         # get component stress
-        nodenum, stress = self.nodal_stress(rnum)
+        nodenum, stress = self.nodal_stress(rnum,nodes)
```
Comment (akaszynski) [inline ansys/mapdl/reader/rst.py:2461]:
```suggestion
        nodenum, stress = self.nodal_stress(rnum, nodes)
```

**Review by akaszynski — APPROVED** [review]

**Comment by akaszynski** [comment]
Nice work!  I made two minor whitespace changes, but other than that good job figuring out how to create a branch and submit a PR!

File: ansys/mapdl/reader/rst.py (hunk 1 above)
Comment (1990chs) [reply ansys/mapdl/reader/rst.py]: Delete the whitespace front and after "=" ?

**Comment by 1990chs** [comment]
> but other than that good job figuring out how to create a branch and submit a PR!
Soory . What does the sentence "but other than that good job figuring out how to create a branch and submit a PR!" mean ?

**Comment by 1990chs** [comment]
> Nice work! I made two minor whitespace changes, but other than that good job figuring out how to create a branch and submit a PR!

Soory . What does the sentence "but other than that good job figuring out how to create a branch and submit a PR!" mean ?

**Comment by akaszynski** [comment]
Thank you for this pull request!

**Comment by 1990chs** [comment]
Very happy to that I can do  some contributions to code

**Pull request merged**